commands/delete-database.go

```diff
@@ -15,81 +17,176 @@ func init() {
 	reset := cli.Command{
 		Name:      "delete",
 		Aliases:   []string{"delete-database"},
-		Usage:     "Delete an imported database",
+		Usage:     "Delete imported database(s)",
 		ArgsUsage: "<database>",
 		Flags: []cli.Flag{
 			forceFlag,
 			configFlag,
+			allFlag,
+			matchFlag,
+			regexFlag,
+			dryRunFlag,
 		},
-		Action: func(c *cli.Context) error {
-			res := resources.InitResources(c.String("config"))
-			db := c.Args().Get(0)
-			if db == "" {
-				return cli.NewExitError("Specify a database", -1)
-			}
-
-			return deleteDatabase(db, res, c.Bool("force"))
-
-		},
+		Action: deleteDatabase,
 	}
 
 	bootstrapCommands(reset)
 }
 
 //deleteDatabase deletes a target database
-func deleteDatabase(database string, res *resources.Resources, forceFlag bool) error {
+func deleteDatabase(c *cli.Context) error {
+	res := resources.InitResources(c.String("config"))
+
+	// Different command flags
+	match := c.String("match")
+	regex := c.String("regex")
+	bulk := c.Bool("all")
+	force := c.Bool("force")
+	dryRun := c.Bool("dry-run")
+	var names []string
+	var err error
+
+	if (match != "") && (!bulk) {
+		// Getting multiple dbs via string search
+		bulk = true
+
+		// Get DB list
+		dbs := res.MetaDB.GetDatabases()
+
+		// Find dbs with matching names
+		for _, db := range dbs {
+			if strings.Contains(db, match) {
+				names = append(names, db)
+			}
+		}
 
-	// get all database names
-	names, err := res.DB.Session.DatabaseNames()
+	} else if (regex != "") && (!bulk) {
+		// Getting multiple dbs via regex query
+		bulk = true
 
-	// check if database exists
-	dbExists := util.StringInSlice(database, names)
+		// Get DB list
+		dbs := res.MetaDB.GetDatabases()
 
-	// check if metadatabase record for database exists
-	mDBExists := util.StringInSlice(database, res.MetaDB.GetDatabases())
+		// Compile regex, check if it's valid
+		regq, err := regexp.Compile(regex)
+		if err != nil {
+			return cli.NewExitError(err.Error, -1)
+		}
+
+		// Find dbs with regex query
+		for _, db := range dbs {
+			matched := regq.MatchString(db)
+			if matched {
+				names = append(names, db)
+			}
+		}
+
+	} else {
+		// get all database names
+		names = res.MetaDB.GetDatabases()
```

Comment:
Could we reflow the code so that if we're deleting a single db (no flags), we skip getting a list of the database names?

Reply:
It might simplify things to treat the single (no flags) case as just another bulk case. You'd end up just processing a list with one item.

I think using the print statements for the bulk case would work for the  single case as well.

Reply:
I can try to run some optimizations, the code flow plan was that it would reach that else statement in case a match or regex flag wasn't set. The code as it was written before gets a list of databases at some point to check if the database exists and this was the adoption of that. I agree though the section between 98 and 138 could be optimized better for this somehow